app/controllers/posts_controller.rb

```diff
@@ -21,6 +21,15 @@ def create
     end
   end
 
+  def destroy
+    @post = Post.find(params[:id])
+    @author = @post.author
+    @post.likes.destroy_all
+    @post.comments.destroy_all
+    @post.destroy
+    redirect_to user_posts_path(@author)
+  end
+
```

Comment:
- [ ] Good job adding authorization rules for the user , however you should include the link between the ability and the postController to ensure that the CanCanCan is used for Post authorization in Post Controller 👍 

```ruby
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```